apps/minifront/src/state/ibc-in.ts

```diff
@@ -109,6 +111,28 @@ const getCompatibleBech32 = (chainName: string, address: Address): string => {
   return bech32Chains.includes(chainName) ? bech32CompatAddress(address) : bech32mAddress(address);
 };
 
+const estimateFee = async ({
+  chainId,
+  client,
+  signerAddress,
+  message,
+}: {
+  chainId: string;
+  client: SigningStargateClient;
+  signerAddress: string;
+  message: EncodeObject;
+}) => {
+  const feeToken = chains.find(({ chain_id }) => chain_id === chainId)?.fees?.fee_tokens[0];
+  const avgGasPrice = feeToken?.average_gas_price;
```

Comment:
Is this the accepted approach? If gas prices suddenly rise, all transactions we initiate will fail?

Reply:
This is what is employed by cosmos kit (using the average price published in the chain registry). Nearly every example of using cosmjs hardcodes this value. However, I share your sentiment. It feels non-flexible. Going to follow up with cosmology team and get their feedback on this. Will merge this for now though.